demo/installer/x86_64/install.sh

```diff
@@ -126,7 +126,11 @@ create_demo_msdos_partition()
 }
 
 eval $create_demo_partition $blk_dev
-demo_dev="${blk_dev}$demo_part"
+if [[${onie_dev} == sd*]]; then
+    demo_dev="${blk_dev}$demo_part"
+else
+    demo_dev="${blk_dev}p$demo_part"
+fi
```

Comment:
I changed it to be more in line with the other tests. Also, mmcblk devices are the only 'special' devices now